system:
  robot_id: 0
  root_suffix: " "
  numberOfCores: 8

  # topic
  pointCloudTopic: "velodyne_points"
  imuTopic: "imu/data"
  imuOdomTopic: "odometry/imu"
  lidarOdomTopic: "odometry/lidar"

factor:
  # lidar
  sensor: "velodyne"         # 'velodyne', 'ouster', 'horizon', 'mid40', 'mid360' or 'avia'
  N_SCAN: 16                 # Velodyne/Ouster: 16, 32, 64, 128; Horizon: 6, mid40: 1, mid360: 4, avia: 8
  Horizon_SCAN: 1800         # Velodyne: 1800; Ouster: 512, 1024, 2048
  lidarMinRange: 2.0         
  lidarMaxRange: 80.0
  mapFrequence: 10

  # imu
  imuFrequence: 150
  imuAccNoise: 3.9939570888238808e-03
  imuGyrNoise: 1.5636343949698187e-03
  imuAccBiasN: 6.4356659353532566e-05
  imuGyrBiasN: 3.5640318696367613e-05
  imuGravity: 9.80511
  imuRPYWeight: 0.01

  # extrinsics, lidar to imu
  extrinsicTrans: [ 0.0, 0.0, 0.0 ]
  extrinsicRot: [ 1, 0, 0,
                  0, 1, 0,
                  0, 0, 1 ]
  extrinsicRPY: [ 1, 0, 0,
                  0, 1, 0,
                  0, 0, 1 ]

feature:
  blind: 0.5
  inf_bound: 10
  group_size: 9
  disA: 0.015              # velodyne/ouster: 0.015, livox: 0.01
  disB: 0.3                # velodyne/ouster: 0.3, livox: 0.1
  p2l_ratio: 400
  limit_maxmid: 9
  limit_midmin: 16
  limit_maxmin: 3.24
  jump_up_limit: 170.0
  jump_down_limit: 8.0
  cos160: 160.0
  edgea: 2.0               # velodyne/ouster: 2.0, livox: 3.0
  edgeb: 0.1               # velodyne/ouster: 0.1, livox: 0.05
  smallp_intersect: 172.5
  smallp_ratio: 1.2
  point_filter_num: 2      # velodyne/ouster: 2, livox: 3

mapping:
  # filter
  mappingCornerLeafSize: 0.2
  mappingSurfLeafSize: 0.2
  globalMapLeafSize: 0.4

  # keyframe
  surroundingkeyframeAddingDistThreshold: 0.5
  surroundingkeyframeAddingAngleThreshold: 0.2
  surroundingkeyframeIntervalTime: 0.1
  surroundingKeyframeSearchRadius: 50.0

